modules/eks/submodules/privatelink/vpc-endpoint-services.tf

```
@@ -37,13 +42,14 @@ resource "aws_lb" "nlbs" {
 resource "aws_lb_target_group" "target_groups" {
   for_each = { for entry in local.listeners : "${entry.service}.${entry.port}" => entry }
 
-  name     = substr("${var.deploy_id}-${each.value.service}-${each.value.port}", -32, -1)
+  name     = "${var.deploy_id}-${random_string.target_group_random_suffix.result}"
```

I believe the error is caused by the offset.

Could we just have:

```suggestion
  name     = substr("${var.deploy_id}-${each.value.service}-${each.value.port}", 0, 32)
```

---

I didnt want to do that because I can have repeat names, for example, for rabbitmq-external there two ports [ 5552,5672 ]

> substr("cloud-seer-bio-rabbitmq-external-5552", 0, 32)
"cloud-seer-bio-rabbitmq-external"
> substr("cloud-seer-bio-rabbitmq-external-5672", 0, 32)
"cloud-seer-bio-rabbitmq-external"

---

that is why I choose to do it in reverse order but in that case during cloud-seer-bio that leads to a name starting with "-" so to avoid other issues changing implementation to <deploy-id> + random

and start with deploy-id is useful to search in aws ui

---

rather than a random string which offers no information, can we just do deploy_id+port, given that the ports are unique

---

Also can you expand on the duplicates, given that the deploy_id is 16 and this is 32, given that the string contains deploy_id + service +port, how can we have a duplicate with at least the deploy_id + 16 characters of the service?

---

I want to avoid other to suffer the same issue in the future relying on the length of service name could be tricky it could work today but not tomorrow

---

the existing service names are "vault-external" and "rabbitmq-external" but someone in the future could include a new service, so it could lead to the same issue

---

regarding port today are different but it someone include another service with the same port that will fail